meta-aos-rpi-install/recipes-core/initrdscripts/files/install_script.sh

```diff
@@ -58,32 +58,60 @@ wait_for_block_device() {
     return 0
 }
 
+update_user() {
+    mkdir -p /sd1 /flash1
+    mount -t auto /dev/mmcblk0p1 /sd1/
+
+    if [ -f /sd1/firstrun.sh ]; then
+        echo "Adding user provided by Raspberry Pi Imager..."
+
+        mount -t auto "/dev/${BLOCK_DEVICE_AOS_PARTITION}" /flash1/
+        output=$(sed -n "s/.*userconf '\(.*\)' '\(.*\)'.*/\1 \2/p" /sd1/firstrun.sh)
+        user=$(echo "$output" | awk '{print $1}')
+
+        if [ -n "$user" ]; then
+            password=$(echo "$output" | awk '{print $2}')
+            sed -i "s/^aos:/$user:/; s/\/home\/aos/\/home\/$user/" "/flash1/etc/passwd"
+            escaped_password=$(printf '%s\n' "$password" | sed 's/[&/\]/\\&/g')
+            sed -i "s/^aos:[^:]*/$user:$escaped_password/" "/flash1/etc/shadow"
+        else
+            echo "No valid user found"
+        fi
+
+        umount /flash1
+    fi
+    
+    umount /sd1
+}
+
 read_args
 
+BLOCK_DEVICE_AOS_PARTITION="${BLOCK_DEVICE}1"
```

Comment:
BLOCK_DEVICE_ROOTFS_PARTITION and rename BLOCK_DEVICE_PARTITION -> BLOCK_DEVICE_AOS_PARTITION